src/librustc/ty/layout.rs

```diff
@@ -267,24 +267,27 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
         }
     }
 
+    fn align_and_pack(&self, repr: &ReprOptions) -> (AbiAndPrefAlign, Option<Align>) {
```

Comment:
What if you added a `pack: Option<Align>` field to `ReprOptions`?
The `Align` encoding is logarithmic so it would occupy less space *and* not need further conversion.

cc @oli-obk @nagisa @rkruppe (curious what other people think)